### PR TITLE
Simpler worker creation

### PR DESCRIPTION
**Pull request by richfitz**
Drops the old form, now everything goes through the configuration. I've changed quite a few defaults too - you could split this PR after the two first commits perhaps to reduce churn?

One thing I've not done here is to rename the default worker configuration from `localhost` to `default`, but I'd like to do that soon too

### CODE REVIEW

File: R/worker.R [hunk 1]
```diff
@@ -64,73 +11,53 @@ rrq_worker <- R6::R6Class(
...
+
       self$id <- worker_id %||% ids::adjective_animal()
-      lockBinding("id", self)
+      self$config <- name_config
+      private$con <- con
       private$keys <- rrq_keys(queue_id, self$id)
-      private$verbose <- verbose
-      private$is_child <- is_child
 
+      ## Check to see if this is a reasonable thing to be doing:
```
Comment (r-ash) [inline R/worker.R:59]: Old comment?

File: R/worker_spawn.R [hunk 2]
```diff
@@ -109,12 +109,12 @@ rrq_worker_manager <- R6::R6Class(
   public = list(
     id = NULL,
 
-    initialize = function(obj, n, logdir = NULL, worker_config = "localhost",
+    initialize = function(obj, n, logdir = NULL, name_config = "localhost",
```
Comment (r-ash) [inline R/worker_spawn.R:112]: Could use the `WORKER_CONFIG_DEFAULT` here and elsewhere?

**Review by r-ash — APPROVED** [review]
Looks good, couple of small comments mostly about TODOs but code look great

File: R/worker_spawn.R (hunk 2 above)
Comment (richfitz) [reply R/worker_spawn.R:112]: I'll check but I was avoiding exposing that constant to the user. But I'd like to move it to be "default" soon, not "localhost"

**Pull request merged**